src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1260,7 +1261,7 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       recordScheduleUpdate();
     }
 
-    if (priorityLevel <= nextPriorityLevel) {
+    if (!isPerformingWork && priorityLevel <= nextPriorityLevel) {
```

Comment:
This extra check ended up not being necessary because we set `nextUnitOfWork = performUnitOfWork` but I left it for clarity. Don't mind removing, though.